### ISSUE TITLE
Check out lgtm.com

### ISSUE THREAD

**Issue by pdxjohnny** [tpm2-software/tpm2-tss#1475]
This looks cool: https://lgtm.com/projects/g/tpm2-software/tpm2-tss (auto analysis in progress as I post this) its a static analysis SaaS

they have a yaml file format we could add the `lgtm.yaml` if the results aren't looking like they picked everything up auto-magically

**Comment by AndreasFuchsSIT** [comment]
Why does it think we're a python project and why does it find only 1 source-code file in here ?
That's weird...

**Comment by diabonas** [comment]
It doesn't analyse the C code because Autoconf Archive is not installed by default in their build environment, but that is easily fixed by adding the following [`.lgtm.yml` file](https://lgtm.com/help/lgtm/lgtm.yml-configuration-file) to our repository, see the [log of the successful analysis](https://lgtm.com/logs/08b5a5b4b922889abe79a0a8e4c5026f03a6726c/lang:cpp):
```
extraction:
  cpp:
    prepare:
      packages: "autoconf-archive"
```

**Comment by diabonas** [comment]
And these are the analysis results for my personal clone of the tpm2-tss repository: https://lgtm.com/projects/g/diabonas/tpm2-tss-test-lgtm/alerts/

**Comment by AndreasFuchsSIT** [comment]
Cool...
But your clone still tagged as python and it only reports the python code Quality... :-(
I guess if we get this fixed, we could add the tag to the readme...

**Comment by AndreasFuchsSIT** [comment]
Ok, we're also A+ in C/C++ code Quality...
Maybe we should add this tag, just to collect tags...
Does anyone know more about the company behind this service ?

**Comment by diabonas** [comment]
> Maybe we should add this tag, just to collect tags...

I have created #1477 to add the configuration and badge. I haven't marked this bug as resolved yet so that we can discuss whether we want to install the LGTM Github  app to [automatically analyse pull requests](https://lgtm.com/help/lgtm/managing-automated-code-review#enabling-pr-integration). This is not absolutely necessary though, new code in the repository will be [automatically analysed once a day](https://lgtm.com/help/lgtm/analysis-faqs#how-often-does-lgtm-check-for-new-commits) anyway.

> Does anyone know more about the company behind this service ?

No idea, they seem to be a rather small company (around 100 employees according to "the Internet"), more like a startup, so I couldn't find out much about them online...

**Comment by diabonas** [comment]
To test that compilation and analysis work as expected and that the repository is reanalysed every day, I [added some bugs on purpose](https://github.com/diabonas/tpm2-tss-test-lgtm/commit/71b934d421247b4a0307d2dedb083463ff556848), which yielded the expected results:
- [Dead code due to goto or break statement](https://lgtm.com/projects/g/diabonas/tpm2-tss-test-lgtm/snapshot/57e1caefcfcb18aa3e243f77a1c82937f8e3a6da/files/src/tss2-sys/api/Tss2_Sys_Load.c#x2e822376822193bc:1)
- [FIXME comment](https://lgtm.com/projects/g/diabonas/tpm2-tss-test-lgtm/snapshot/57e1caefcfcb18aa3e243f77a1c82937f8e3a6da/files/src/util/log.c#x595e6ec3b0c4b53f:1)

**Comment by flihp** [comment]
How is the login to this site managed? Is their model for authorization tied to the github org permissions or is there a separate login / password etc?

**Comment by pdxjohnny** [comment]
GitHub org oauth

**Comment by diabonas** [comment]
For the normal daily code analysis, no permissions are required, they pull the code from Github themselves (in fact there doesn't seem to be a way to stop them from doing this, they automatically analyse "a large number of open source projects" according to their [privacy policy](https://lgtm.com/privacy)). The only thing we would need to do is to add the `.lgtm.yml` to our repository so that the code can be analysed correctly, which currently fails due to missing dependencies.

For the instant analysis of the code in new pull requests, a Github app needs to be installed for the repository, which I assume can only be done by administrators of the organisation.

You can login into the LGTM website using a Github account as well, but that doesn't seem to do much: you can watch repositories by adding them to a custom "My projects" lists and click on the "Enable PR code reviews" button, which takes you to the Github page for installing the LGTM Github app.

**Comment by flihp** [comment]
merged in #1477
